File: src/main/resources/static/js/app/UserService.js
'use strict'

angular.module('demo.services', []).factory('UserService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getUserById = function(userId) {
				var url = CONSTANTS.getUserByIdUrl + userId;
				return $http.get(url);
			}
			service.getAllUsers = function() {
				return $http.get(CONSTANTS.getAllUsers);
			}
			service.saveUser = function(userDto) {
				return $http.post(CONSTANTS.saveUser, userDto);
			}
			service.saveUserBalance = function(userDto) {
				return $http.post(CONSTANTS.saveUserBalance, userDto);
			}
			service.withdrawFunds = function (userDto, withdrawal) {
				var url = CONSTANTS.withdrawFundsUrl + withdrawal;
				return $http.post(url, userDto);
			}
			service.depositFunds = function (userDto, deposit) {
				var url = CONSTANTS.depositFundsUrl + deposit;
				return $http.post(url, userDto);
			}
			return service;
		} ]);

File: src/main/java/com/example/simpleatm/service/UserService.java
package com.example.simpleatm.service;

import java.util.List;

import com.example.simpleatm.dto.UserDto;

/**
 * Created by ashish on 13/5/17.
 */
public interface UserService {
    UserDto getUserById(Integer userId);
    void saveUser(UserDto userDto);
    void saveUserBalance(UserDto userDto);
    Float withdrawFunds(UserDto userDto, Float withdrawal);
    Float depositFunds(UserDto userDto, Float deposit);
    List<UserDto> getAllUsers();
}
